src/cart/tests_common.h

```diff
@@ -104,6 +108,35 @@ tc_progress_stop(void)
 	opts.shutdown = 1;
 }
 
+/* Write a completion file to signal graceful server shutdown */
+void
+write_completion_file(void)
+{
+	FILE	*fptr;
+	char	*dir;
+	char	*completion_file;
+	char	*tmp_str;
+	char	pid[6];
+	pid_t	_pid;
+
+	_pid = getpid();
+	sprintf(pid, "%d", _pid);
+
+	dir = getenv("DAOS_TEST_SHARED_DIR");
+	tmp_str = strcat(dir, "/test-servers-completed.txt.");
+	completion_file = strcat(tmp_str, pid);
+
+	fptr = fopen(completion_file, "w");
+	DBG_PRINT("Opened completion file to write: %s.\n", completion_file);
+	if (fptr == NULL) {
+		D_ERROR("Error opening completion file for writing.\n");
+		return;
```

Comment:
we should assert here

Reply:
984ab8326148b1094ecf16bd2e180f90ebf80c23